pkg/validator/validate.go

```diff
@@ -0,0 +1,189 @@
+package validator
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/treeverse/lakefs/pkg/graveler"
```

Comment:
agree with @nopcoder, if we are creating this external validator it will be great to make it package free. I think that you decided to move validatePath to the catalog package for the same reason.

Reply:
done, it's now package free.